### ISSUE TITLE
elfeed search menu updated after every new feed is parsed

### ISSUE THREAD

**Issue by justmytwospence** [skeeto/elfeed#151]
I have a lot of feeds, so I like to start working my way through them even before they are all done updating. This is made difficult however, because every time the search menu updates to accomodate a newly fetched feed, all the feeds that I have marked as unread also update, meaning they disappear, causing the search menu to change radically and often causing me to archive things I don't mean to, etc. I understand that this is a fairly deep seated problem due to the way that elfeed treats unread posts as just another tag. Are there any potential workarounds though? Being able to effectively work my way through feeds while they are updating would really mitigate the amount of time it takes to update 400+ feeds (I probably wouldn't even notice it at that point).

**Comment by skeeto** [comment]
(Replying again since yesterday's GitHub downtime swallowed the last one.)

Elfeed is layered like an onion, with the database at the middle (elfeed-db.el), with the code to populate it built on top (elfeed.el), and on the interface layer (elfeef-search.el, elfeed-show.el) as the surface. The interface is essentially an extension to the fetching and database layers below it. It even uses hooks from these layers. As such, you can disconnect the automatic search buffer refresh/redraw by removing `elfeed-search-update` from `elfeed-update-hooks`.

Here's where it gets a little messy: You'll need undo this *after* elfeed-search has loaded, which means you'll need to do it in its mode hook.

~~~el
(add-hook 'elfeed-search-mode-hook
          (lambda ()
            (remove-hook 'elfeed-update-hooks
                         'elfeed-search-update)))
~~~

However, this will also disable updating the header during feed updates.

**Comment by justmytwospence** [comment]
Yeah, not having any of the headers update defeats the purpose of this attempt anyway. I'm pretty sure there is no way to do what I want to do with the way elfeed is currently designed. It would need to add the ability to do partial updates that ignore specific tags (unread specifically). Alternatively, maybe if I could figure out a keybinding that pseudo-removes the the unread tag by flagging that entry somehow, but then doesn't actually remove the unread tag until immediately before executing elfeed-search-update. That second one is actually seeming like my best bet for working around this. Thoughts?
